modules/vstudio/vs2010_vcxproj_filters.lua

```diff
@@ -47,18 +47,55 @@
 
 --
 -- The first portion of the filters file assigns unique IDs to each
--- directory or virtual group. Would be cool if we could automatically
--- map vpaths like "**.h" to an <Extensions>h</Extensions> element.
+-- directory or virtual group.
 --
 
 	function m.uniqueIdentifiers(prj)
-		local tr = project.getsourcetree(prj)
+		local knownFilters = {
+			['Source Files'] = {
+				order = 1,
+				extensions = 'cpp;c;cc;cxx;def;odl;idl;hpj;bat;asm;asmx',
+			},
+			['Header Files'] = {
+				order = 2,
+				extensions = 'h;hpp;hxx;hm;inl;inc;xsd',
+			},
+			['Resource Files'] = {
+				order = 3,
+				extensions = 'rc;ico;cur;bmp;dlg;rc2;rct;bin;rgs;gif;jpg;jpeg;jpe;resx;tiff;tif;png;wav;mfcribbon-ms',
+			},
+		}
+
+		local sortInFilterOrder = function (a,b)
+			if #a.children > 0 or #b.children > 0 then
+				local filterA = knownFilters[a.name]
+				local filterB = knownFilters[b.name]
+				if filterA ~= nil and filterB ~= nil then
+					return filterA.order < filterB.order
+				end
+				if filterA ~= nil and filterB == nil then
+					return true
+				end
+				if filterA == nil and filterB ~= nil then
+					return false
```

Comment:
Possibly:
```suggestion
				local unknown_filter = { order = 4 }
				local filterA = knownFilters[a.name] or unknown_filter 
				local filterB = knownFilters[b.name] or unknown_filter 

				return filterA.order < filterB.order
```

Reply:
Nice, thanks! Sorry for my "baby Lua", unfortunately Lua is not a language I'm really familiar/comfortable with.

If this PR is accepted, I can apply this suggestion (perhaps with `order = 999` instead of `4`?) or you could apply it when merging, of course.